src/poller.ts

```diff
@@ -0,0 +1,69 @@
+import type { Options } from './types'
+
+export default class Poller {
+  private timeoutId: number
+  private isRunning: boolean
+
+  constructor(
+    private fetchFlagsFn: () => Promise<any>,
+    private configurations: Options,
+    private pollInterval: number,
+    private maxAttempts: number
+  ) {}
+
+  public start(): void {
+    if (this.isPolling()) {
+      this.logDebug('Polling was requested but is already running - only one poller can run at a time.')
+      return
+    }
+
+    this.isRunning = true
+
+    // Don't start polling immediately as we have alreadu fetched flags on client initialization
+    window.setTimeout(() => this.poll(), this.configurations.pollingInterval)
+  }
+
+  private poll(): void {
+    this.attemptFetch().finally(() => {
+      this.timeoutId = window.setTimeout(() => this.poll(), this.configurations.pollingInterval)
+    })
+  }
+
+  private async attemptFetch(): Promise<void> {
+    for (let i = 1; i <= this.maxAttempts; i++) {
```

Comment:
Instead of immediately polling again, it should wait a bit and try again. The streaming implementation has this in a very simplistic form; can it be extracted into a utility function that either could user? We'll need to double back and make sure it has some form of exponential back off, but the random 1-10 second retry is enough for now. This was a big concern for the BE team as clients would potentially fail and then all ping back at the same time, effectively DDOSing the service when it probably went down from being over worked in the first place.

Reply:
I've introduced a delay to each attempt. I didn't end up extracting a utility function for the reconnect specifically - but I did extract the function for calculating the delay. I've updated the test cases. 
Thanks for catching this.